bundles/binding/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/SagerCasterBindingProvider.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sagercaster;
+
+import org.openhab.binding.sagercaster.internal.CommandType;
+import org.openhab.core.binding.BindingProvider;
+
+/**
+ * @author Gaël L'hopital
+ * @since 1.7.0
+ */
+public interface SagerCasterBindingProvider extends BindingProvider {
+
+	/**
+	 * @return all the binding config matching a given <code>commandType</code>
+	 */
+	public Iterable<String> getAllItemNames(CommandType commandType);
```

Comment:
since the Set of Items is filtered by commandType i would suggest to rename this method to something like "getItemNamesBy()" or alike